frontend/src/components/wallet/LnurlPay.tsx

```diff
@@ -0,0 +1,124 @@
+import { LightningAddress } from "@getalby/lightning-tools";
+import React from "react";
+import { Button } from "src/components/ui/button";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useToast } from "src/components/ui/use-toast";
+import PaymentSuccessCard from "src/components/wallet/PaymentSuccessCard";
+
+import { PayInvoiceResponse } from "src/types";
+import { request } from "src/utils/request";
+
+type LnurlPayProps = {
+  lnAddress: LightningAddress;
+  onReset: () => void;
+  onAmountChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+function LnurlPay({ lnAddress, onReset, onAmountChange }: LnurlPayProps) {
+  const [amount, setAmount] = React.useState("");
+  const [comment, setComment] = React.useState("");
+  const [preimage, setPreimage] = React.useState("");
+  const [isLoading, setLoading] = React.useState(false);
+  const { toast } = useToast();
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      setLoading(true);
+      const invoice = await lnAddress.requestInvoice({
```

Comment:
is there a way to not duplicate the invoice payment logic? you already pass callbacks as props like onReset, could there be one to pay an invoice?